### PR TITLE
feat: DID creator for service packages

### LINKED ISSUE #322: DID method to be used during DID Exchange
DID Exchange service need to provide DID method name for creating DID.

belongs to #42

### PR DESCRIPTION
**Pull request by sudeshrshetty**
- added DID creator which hides functionality of passing DID method to
wallet DID creator
- closes #322

### CODE REVIEW

File: pkg/didcomm/protocol/didexchange/service_test.go [hunk 1]
```diff
@@ -517,9 +516,16 @@ func (p *mockProvider) OutboundDispatcher() dispatcher.Outbound {
 	return &mockdispatcher.MockOutbound{}
 }
 
-// CryptoWallet returns the pack wallet service
-func (p *mockProvider) DIDWallet() wallet.DIDCreator {
-	return &mockwallet.CloseableWallet{CreateDid: &did.Doc{
+type mockDIDCreator struct {
```
Comment (rolsonquadras) [inline pkg/didcomm/protocol/didexchange/service_test.go]: Can all these mockDIDCreator be consolidated in one place? Similar change is being done for DIDExchange mock service in #348 .

Comment (sudeshrshetty) [reply pkg/didcomm/protocol/didexchange/service_test.go]: done

File: pkg/client/didexchange/client_test.go [hunk 2]
```diff
@@ -10,6 +10,8 @@ import (
 	"fmt"
 	"testing"
 
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
```
Comment (troyronda) [inline pkg/client/didexchange/client_test.go]: imports are not grouped.

Comment (sudeshrshetty) [reply pkg/client/didexchange/client_test.go]: done

File: pkg/framework/aries/default.go [hunk 3]
```diff
@@ -9,6 +9,8 @@ package aries
 import (
 	"fmt"
 
+	"github.com/hyperledger/aries-framework-go/pkg/common/did"
```
Comment (fqutishat) [inline pkg/framework/aries/default.go]: wrong location

Comment (sudeshrshetty) [reply pkg/framework/aries/default.go]: done

Comment (sudeshrshetty) [reply pkg/framework/aries/default.go]: done

**Pull request merged**